experimental/handoff-library/csharp_dotnetcore/Microsoft.Bot.Builder.Handoff/HandoffHttpSupport.cs

```diff
@@ -0,0 +1,235 @@
+using Microsoft.Rest;
+using Newtonsoft.Json;
+
+// This file mimics the generated code in Conversations.cs which disables the following SA warnings:
+#pragma warning disable SA1629
+#pragma warning disable SA1513
+#pragma warning disable SA1515
+#pragma warning disable SA1312
+#pragma warning disable SA1122
+#pragma warning disable SA1507
+#pragma warning disable SA1508
+
+namespace Microsoft.Bot.Builder.Handoff
+{
+    /// <summary>
+    /// Making HTTP calls for handoff. When handoff is integrated
+    /// into the SDK proper, this code migrates to Conversations.cs.
+    /// </summary>
+    public static class HandoffHttpSupport
+    {
+        public static async Task<HttpOperationResponse<ResourceResponse>> HandoffWithHttpMessagesAsync(IServiceOperations<ConnectorClient> conversations, string conversationId, HandoffParameters handoffParameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (conversationId == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "conversationId");
+            }
+            if (handoffParameters == null)
+            {
+                throw new ValidationException(ValidationRules.CannotBeNull, "handoffParameters");
+            }
+
+            // Construct URL
+            var _baseUrl = conversations.Client.BaseUri.AbsoluteUri;
+            var _url = new System.Uri(new System.Uri(_baseUrl + (_baseUrl.EndsWith("/") ? "" : "/")), "v3/conversations/{conversationId}/handoff").ToString();
+            _url = _url.Replace("{conversationId}", System.Uri.EscapeDataString(conversationId));
+            // Create HTTP transport objects
+            var _httpRequest = new HttpRequestMessage();
+            HttpResponseMessage _httpResponse = null;
+            _httpRequest.Method = new HttpMethod("POST");
+            _httpRequest.RequestUri = new System.Uri(_url);
+            // Set Headers
+            if (customHeaders != null)
+            {
+                foreach (var _header in customHeaders)
+                {
+                    if (_httpRequest.Headers.Contains(_header.Key))
+                    {
+                        _httpRequest.Headers.Remove(_header.Key);
+                    }
+                    _httpRequest.Headers.TryAddWithoutValidation(_header.Key, _header.Value);
+                }
+            }
+
+            // Serialize Request
+            string _requestContent = null;
+            if (handoffParameters != null)
+            {
+                _requestContent = Rest.Serialization.SafeJsonConvert.SerializeObject(handoffParameters, conversations.Client.SerializationSettings);
+                _httpRequest.Content = new StringContent(_requestContent, System.Text.Encoding.UTF8);
+                _httpRequest.Content.Headers.ContentType = System.Net.Http.Headers.MediaTypeHeaderValue.Parse("application/json; charset=utf-8");
```

Comment:
How do we know UTF-8 is the right encoding here? It's not been set anywhere. This may have unicode / emoji issues as a result of a potential mismatch.

Reply:
This is the encoding used for all other activities. It handles Unicode including emojis (as long as they are Unicode).